synapse/handlers/register.py

```diff
@@ -163,7 +163,9 @@ def register_user(
         """
         yield self.check_registration_ratelimit(address)
 
-        yield self.auth.check_auth_blocking(threepid=threepid)
+        # do not check_auth_blocking if the call is coming through the Admin API
+        if address:
```

Comment:
this is not reliable. There are other code paths which will set `address=None`.

I think the easiest solution is to add yet another parameter to `register_user` indicating whether the request came from the admin API, though I have to say that refactoring it into smaller functions is also compelling...